x-pack/legacy/plugins/siem/public/containers/detection_engine/rules/api.ts

```diff
@@ -223,3 +228,83 @@ export const createPrepackagedRules = async ({ signal }: BasicFetchProps): Promi
   await throwIfNotOk(response);
   return true;
 };
+
+/**
+ * Imports rules in the same format as exported via the _export API
+ *
+ * @param fileToImport File to upload containing rules to import
+ * @param overwrite whether or not to overwrite rules with the same ruleId
+ * @param signal AbortSignal for cancelling request
+ *
+ * @throws An error if response is not OK
+ */
+export const importRules = async ({
+  fileToImport,
+  overwrite = false,
+  signal,
+}: ImportRulesProps): Promise<ImportRulesResponse> => {
+  const formData = new FormData();
+  formData.append('file', fileToImport);
+
+  const response = await fetch(
+    `${chrome.getBasePath()}${DETECTION_ENGINE_RULES_URL}/_import?overwrite=${overwrite}`,
+    {
+      method: 'POST',
+      credentials: 'same-origin',
+      headers: {
+        'kbn-xsrf': 'true',
+      },
+      body: formData,
+      signal,
+    }
+  );
+
+  await throwIfNotOk(response);
+  return response.json();
+};
+
+/**
+ * Export rules from the server as a file download
+ *
+ * @param excludeExportDetails whether or not to exclude additional details at bottom of exported file (defaults to false)
+ * @param filename of exported rules. Be sure to include `.ndjson` extension! (defaults to localized `rules_export.ndjson`)
+ * @param ruleIds array of rule_id's (not id!) to export (empty array exports _all_ rules)
+ * @param signal AbortSignal for cancelling request
+ *
+ * @throws An error if response is not OK
+ */
+export const exportRules = async ({
+  excludeExportDetails = false,
+  filename = `${i18n.EXPORT_FILENAME}.ndjson`,
+  ruleIds = [],
+  signal,
+}: ExportRulesProps): Promise<Blob> => {
+  const body =
+    ruleIds.length > 0
+      ? JSON.stringify(
+          ruleIds.reduce<ExportRulesPayload>(
+            (acc, ruleId) => ({ objects: [...acc.objects, { rule_id: ruleId }] }),
+            { objects: [] }
+          )
+        )
+      : undefined;
```

Comment:
Isn't this just a map like so?

```ts
  const body =
    ruleIds.length > 0
      ? JSON.stringify({
          objects: ruleIds.map(rule => ({ rulde_id: rule })),
        })
      : undefined;
```

You're not reducing the number of things but rather just transforming?